clients/da-vinci-client/src/main/java/com/linkedin/davinci/kafka/consumer/StoreIngestionTask.java

```diff
@@ -2451,6 +2451,17 @@ private int internalProcessConsumerRecord(
             partitionConsumptionState,
             leaderProducedRecordContext,
             currentTimeMs);
+        if (isHybridMode() && partitionConsumptionState.hasLagCaughtUp()) {
+          long afterProcessingRecordTimestampNs = System.currentTimeMillis();
+          long brokerProducedTimeStamp = (leaderProducedRecordContext == null)
+              ? kafkaValue.producerMetadata.messageTimestamp
```

Comment:
Could you please leave a comment here calling out that for follower replicas, if we add this latency with the producer to local broker latency, we would double count the leader producer to acknowledge latency?
And we can also leave a TODO message saying that everything will be cleaned up after upgrading the KME protocol to carry the producer timestamp end to end.